icu4c/source/i18n/listformatter.cpp

```diff
@@ -348,6 +348,7 @@ const ListFormatInternal* ListFormatter::getListFormatInternal(
     return result;
 }
 
+#if !UCONFIG_NO_FORMATTING
```

Comment:
Why do you need to block out this method?  Because it depends on the enum symbols from ulistformatter.h?  It doesn't make sense that this method is blocked but not the others on this class.  Is the more correct fix to pull the enums in ulistformatter.h *out of* UCONFIG_NO_FORMATTING?  I don't know the answer to that.

Reply:
Updated to conditionally remove the method and the call. Moving the enums in ulistformatter.h doesn't seem right because everything else is in that file is conditioned on UCONFIG_NO_FORMATTING.

Reply:
This doesn't compile if UNCONFIG_NO_FORMATTING = 1.  The problem is that this is only used in the previous call using the new interface. I'm updating this to put both the call and this function under UCONFIG_NO_FORMATTING.